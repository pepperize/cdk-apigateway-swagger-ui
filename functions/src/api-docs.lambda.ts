import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda/trigger/api-gateway-proxy";
import { APIGateway, AWSError } from "aws-sdk";
import { OpenApiSpecPatcher } from "./open-api-spec-patcher";

let apiGateway: APIGateway;

const patcher = new OpenApiSpecPatcher({ pathsToRemove: ["/api-docs.json", "/api-docs/{proxy+}", "/api-docs"] });

export async function handler(request: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
  apiGateway = new APIGateway();

  try {
    const spec = await apiGateway
      .getExport({
        restApiId: request.requestContext.apiId,
        stageName: request.requestContext.stage,
        exportType: "oas30",
        parameters: {},
        accepts: "application/json",
      })
      .promise();

    const patchedSpec = patcher.patchJson(spec.body!.toString());

    return {
      statusCode: 200,
      headers: { "Content-Type": "application/json", "Content-Disposition": "inline" },
      body: patchedSpec,
      isBase64Encoded: false,
    };
  } catch (e) {
    const error = e as AWSError;
    console.log(error);

    return {
      statusCode: 500,
      headers: { "Content-Type": "application/problem+json" },
      body: JSON.stringify({
        type: "about:blank",
        title: error.code,
        status: 500,
        description: error.message,
      }),
      isBase64Encoded: false,
    };
  }
}
