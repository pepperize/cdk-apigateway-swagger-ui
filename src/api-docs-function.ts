// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
import * as path from 'path';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import { Construct } from 'constructs';

/**
 * Props for ApiDocsFunction
 */
export interface ApiDocsFunctionProps extends lambda.FunctionOptions {
}

/**
 * An AWS Lambda function which executes functions/src/api-docs.
 */
export class ApiDocsFunction extends lambda.Function {
  constructor(scope: Construct, id: string, props?: ApiDocsFunctionProps) {
    super(scope, id, {
      description: 'functions/src/api-docs.lambda.ts',
      ...props,
      runtime: new lambda.Runtime('nodejs18.x', lambda.RuntimeFamily.NODEJS),
      handler: 'index.handler',
      code: lambda.Code.fromAsset(path.join(__dirname, '../assets/functions/src/api-docs.lambda')),
    });
    this.addEnvironment('AWS_NODEJS_CONNECTION_REUSE_ENABLED', '1', { removeInEdge: true });
  }
}
