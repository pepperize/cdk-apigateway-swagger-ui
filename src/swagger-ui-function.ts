// ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
import * as path from 'path';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import { Construct } from 'constructs';

/**
 * Props for SwaggerUiFunction
 */
export interface SwaggerUiFunctionProps extends lambda.FunctionOptions {
}

/**
 * An AWS Lambda function which executes functions/src/swagger-ui.
 */
export class SwaggerUiFunction extends lambda.Function {
  constructor(scope: Construct, id: string, props?: SwaggerUiFunctionProps) {
    super(scope, id, {
      description: 'functions/src/swagger-ui.lambda.ts',
      ...props,
      runtime: new lambda.Runtime('nodejs18.x', lambda.RuntimeFamily.NODEJS),
      handler: 'index.handler',
      code: lambda.Code.fromAsset(path.join(__dirname, '../assets/functions/src/swagger-ui.lambda')),
    });
    this.addEnvironment('AWS_NODEJS_CONNECTION_REUSE_ENABLED', '1', { removeInEdge: true });
  }
}
