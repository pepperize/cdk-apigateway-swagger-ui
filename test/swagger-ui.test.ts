import { Stack } from "aws-cdk-lib";
import { RestApi } from "aws-cdk-lib/aws-apigateway";
import { SwaggerUi } from "../src";
import "jest-cdk-snapshot";

describe("SwaggerUi", () => {
  it("Should match snapshot", () => {
    // Given
    const stack = new Stack();
    const api = new RestApi(stack, "Api");

    // When
    new SwaggerUi(stack, "SwaggerUi", { resource: api.root });

    // Then
    expect(stack).toMatchCdkSnapshot({
      ignoreAssets: true,
      ignoreCurrentVersion: true,
      ignoreMetadata: true,
      ignoreTags: true,
    });
  });
});
