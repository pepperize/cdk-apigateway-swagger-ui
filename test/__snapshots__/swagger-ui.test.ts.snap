// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwaggerUi Should match snapshot 1`] = `
Object {
  "Outputs": Object {
    "ApiEndpoint4F160690": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "ApiF70053CD",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "ApiDeploymentStageprod3EB9684E",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "ApiAccountA18C9B29": Object {
      "DependsOn": Array [
        "ApiF70053CD",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "ApiCloudWatchRole73EC6FC4",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "ApiCloudWatchRole73EC6FC4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ApiDeploymentB17BE62D53ff4e9ae5ff9cf54bd6b3125f8b416f": Object {
      "DependsOn": Array [
        "ApiapidocsjsonGETFBE5FD3D",
        "ApiapidocsjsonF4C2B6CE",
        "ApiapidocsproxyGET9E5CEF71",
        "Apiapidocsproxy08FBC5FA",
        "ApiapidocsEBBEE27E",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "ApiDeploymentStageprod3EB9684E": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "ApiDeploymentB17BE62D53ff4e9ae5ff9cf54bd6b3125f8b416f",
        },
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "ApiF70053CD": Object {
      "Properties": Object {
        "Name": "Api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "ApiapidocsEBBEE27E": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "ApiF70053CD",
            "RootResourceId",
          ],
        },
        "PathPart": "api-docs",
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ApiapidocsjsonF4C2B6CE": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "ApiF70053CD",
            "RootResourceId",
          ],
        },
        "PathPart": "api-docs.json",
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ApiapidocsjsonGETApiPermissionApiGETapidocsjson0EB268C0": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "SwaggerUiApiDocsFunctionD75F49C7",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiF70053CD",
              },
              "/",
              Object {
                "Ref": "ApiDeploymentStageprod3EB9684E",
              },
              "/GET/api-docs.json",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiapidocsjsonGETApiPermissionTestApiGETapidocsjson7A485C46": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "SwaggerUiApiDocsFunctionD75F49C7",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiF70053CD",
              },
              "/test-invoke-stage/GET/api-docs.json",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiapidocsjsonGETFBE5FD3D": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "SwaggerUiApiDocsFunctionD75F49C7",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "ApiapidocsjsonF4C2B6CE",
        },
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "Apiapidocsproxy08FBC5FA": Object {
      "Properties": Object {
        "ParentId": Object {
          "Ref": "ApiapidocsEBBEE27E",
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "ApiapidocsproxyGET9E5CEF71": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "SwaggerUiSwaggerUiFunction76567BED",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "Apiapidocsproxy08FBC5FA",
        },
        "RestApiId": Object {
          "Ref": "ApiF70053CD",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "ApiapidocsproxyGETApiPermissionApiGETapidocsproxyF5189CF8": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "SwaggerUiSwaggerUiFunction76567BED",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiF70053CD",
              },
              "/",
              Object {
                "Ref": "ApiDeploymentStageprod3EB9684E",
              },
              "/GET/api-docs/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "ApiapidocsproxyGETApiPermissionTestApiGETapidocsproxy8A8A4E69": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "SwaggerUiSwaggerUiFunction76567BED",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "ApiF70053CD",
              },
              "/test-invoke-stage/GET/api-docs/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "b120b13d9d868c7622e7db1b68bae4c0f82ffd0227b8c15f2cef38e186ff3827.zip",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SwaggerUiApiDocsFunctionD75F49C7": Object {
      "DependsOn": Array [
        "SwaggerUiApiDocsFunctionServiceRoleDefaultPolicyA6BC6934",
        "SwaggerUiApiDocsFunctionServiceRoleFFB489F6",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "3220ea8279d3101425d2fbce1a3f96416a70702a1e9c8b6c88dca5532d8d6cb8.zip",
        },
        "Description": "functions/src/api-docs.lambda.ts",
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "SwaggerUiApiDocsFunctionServiceRoleFFB489F6",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "SwaggerUiApiDocsFunctionLogRetention78C079DF": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "SwaggerUiApiDocsFunctionD75F49C7",
              },
            ],
          ],
        },
        "RetentionInDays": 7,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "SwaggerUiApiDocsFunctionServiceRoleDefaultPolicyA6BC6934": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "apigateway:GET",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":apigateway:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    "::/restapis/",
                    Object {
                      "Ref": "ApiF70053CD",
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "SwaggerUiApiDocsFunctionServiceRoleDefaultPolicyA6BC6934",
        "Roles": Array [
          Object {
            "Ref": "SwaggerUiApiDocsFunctionServiceRoleFFB489F6",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SwaggerUiApiDocsFunctionServiceRoleFFB489F6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SwaggerUiSwaggerUiFunction76567BED": Object {
      "DependsOn": Array [
        "SwaggerUiSwaggerUiFunctionServiceRoleD19EE363",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "3fcf7b4fa192ea77a55f7824279b84621d2ad07e507f98e2194a6c5a679584a6.zip",
        },
        "Description": "functions/src/swagger-ui.lambda.ts",
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "SwaggerUiSwaggerUiFunctionServiceRoleD19EE363",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "SwaggerUiSwaggerUiFunctionLogRetentionF848B51E": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "SwaggerUiSwaggerUiFunction76567BED",
              },
            ],
          ],
        },
        "RetentionInDays": 7,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "SwaggerUiSwaggerUiFunctionServiceRoleD19EE363": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
